### ISSUE TITLE
Sub tasks are not moved when start of parent task changes

### ISSUE THREAD

**Issue by alexanderopdeweegh** [bardsoftware/ganttproject#1771]
Hi,

I've created a task which contains a number child tasks. Then, I've edited the properties of the parent task and have set the "start no earlier than" date. I expect that not only the parent task shifts towards this starting date (which does happen correctly), but also the child tasks should shift in time. But this does not happen which results in child tasks beginning earlier than their parent task...

Hope this bug can be fixed.

Regards,
Alexander

**Comment by dbarashev** [comment]
Hello,

> set the "start no earlier than" date

did you mean "earliest begin"? In my test changing earliest begin date in a summary task just fails with error (which is also a bug of course), not changing any dates at all.

As a workaround, if you intention is to make sure that summary task can't start earlier than some specified date, you can create a milestone at that date and draw a FS dependency between the milestone and the summary task. See the screenshot:
![image](https://user-images.githubusercontent.com/2028330/92906351-13232f00-f42d-11ea-8e5c-3a92b3ae73c8.png)

**Comment by dbarashev** [comment]
I changed the issue title because child tasks do change their dates when summary task moves (e.g. because of dependencies). The problem shows up particularly with the earliest begin constraint.

**Comment by alexanderopdeweegh** [comment]
Yes, I mean the earliest begin setting. I will use the workaround until the bug is fixed. Thanks for your quick response.
